### ISSUE TITLE
Add support for distance matrices

### ISSUE THREAD

**Issue by xrotwang** [cldf/cldf#88]
TODO

**Comment by LinguList** [comment]
Agree, also given that we have different parameters (maybe you will have a more elegant way to model this....):

parameter | language A | language B | value
--- | --- | --- | ---
shared-vocabulary | German | English | 0.8
lexicostatistical_disttance | German | English 0.2
mutual_intellibigility | German | English | 0.6
intelligibility | German | English | 0.7
intelligibility | English | German | 0.6
geographic-distance | German | English | 500 km

Furthermore, we could even annotate, if possible, WHAT parameter we used, e.g., for shared vocabulary: Swadesh-1952-200, Swadesh-1955-100, etc.

**Comment by LinguList** [comment]
This is not to say that we have those exact params everywhere, but in fact this is what the differences in the literature are often about, the distances differ due to:

1. the concept list that was used
2. the dataset that was used
3. the measure (jaccard, hamming of binary cognates, Swadesh-style calculation, ToB calculation)

**Comment by xrotwang** [comment]
Ok, good examples. So there is good reason to add special support for bivariate functions - without going to multivariate > 2 right away, correct?

**Comment by LinguList** [comment]
Yes, I'd say so, although, for sound correspondence patterns, we'd have multi-variate, but these woudl anyway not have the same value for n languages (this is more similar to cognate sets).

**Comment by SimonGreenhill** [comment]
Yes, I think bivariate is sensible, and multivariate is a problem for another day.

**Comment by xrotwang** [comment]
Then I'd say the most pragmatic encoding would be a CLDF component for bi-variate parameters with
- a required `BivariateValueTable`
- an optional `BivariateParameterTable`, adding metadata about the parameters.

**Comment by SimonGreenhill** [comment]
Sounds good!

**Comment by RustyGray** [comment]
Around? Just discussed SLE plans with Mattis but could find you. R.

Russell Gray
Director Max Planck Institute for the Science of Human History
Kahlaische Strasse 10
D-07745 Jena, GERMANY
TEL: +49-3641-68 68 01
FAX: +49-3641-68 68 68
Departmental Administrator  schueck@shh.mpg.de<mailto:schueck@shh.mpg.de>
http://www.shh.mpg.de/2375/en
http://scholar.google.co.nz/citations?user=FzE-34YAAAAJ&hl=en
Database of places, language, culture & environment: www.d-place.org

On 15. Oct 2018, at 14:31, Simon J Greenhill <notifications@github.com<mailto:notifications@github.com>> wrote:


Sounds good!

**Comment by xrotwang** [comment]
@SimonGreenhill @LinguList do we have any concrete examples for this, i.e. datasets where such data is readily available - for testing and as examples?

**Comment by LinguList** [comment]
Distance matrices are available for many Chinese datasets, I started to
collect them:
https://github.com/digling/cddb/tree/master/datasets/Tang2007/raw (some
more in the CDDB: https://github.com/digling/cddb)

Tree samples plus data in cldf are available, e.g., in
https://github.com/lingpy/sino-tibetan-paper

**Comment by xrotwang** [comment]
Here's another set of distance matrices which would profit from CLDF being able to model these efficiently: https://doi.org/10.5281/zenodo.7973820

> Euclide, the crow, the wolf and the pedestrian: distance metrics for linguistic typology - Dataset, Guzman Naranjo and Jäger 2023

**Comment by LinguList** [comment]
Yes! Should we start by addi g examples to cldf-datasets? Or make a new org called language-distances or siimilar and assemble data there?

**Comment by xrotwang** [comment]
I think cldf-datasets is fine.
